docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,405 @@
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-01-02T03:04:05Z"
+  }
+}
+```
+
+### Relation of the Artifact to Image Manifest
+
+```text
+ Image Index                                                 
+ +----------------------------------------------------------+
+ |                                                          |
+ |  Image Manifest A             Image Manifest B           |
+ |  +-----------------------+    +-----------------------+  |
+ |  |platform:              |    |platform:              |  |
+ |  |  archictecture: amd64 |    |  archictecture: arm   |  |
+ |  |  os: linux            |    |  os: linux            |  |
+ |  |                       |    |                       |  |
+ |  |digest: abc123         |    |digest: def456         |  |
+ |  +--------------------|--+    +--------------------|--+  |
+ |                       |                            |     |
+ +-----------------------|----------------------------|-----+
+                         |                            |      
+                         |                            |      
+    Artifact A           |       Artifact B           |      
+    +--------------------|--+    +--------------------|--+   
+    |                    |  |    |                    |  |   
+    | Subject Descriptor |  |    | Subject Descriptor |  |   
+    | +------------------|-+|    | +------------------|-+|   
+    | | digest: abc123     ||    | | digest: def456     ||   
+    | +--------------------+|    | +--------------------+|   
+    |                       |    |                       |   
+    +-----------------------+    +-----------------------+   
+```
+
+The design imposes a strong 1 to 1 relationship between the compatibility artifact and the image.
+That allows users to independently release artifacts and attach compatibility to the already existing images.
+
+The disadvantage of this solution is that you cannot create compatibility that points to multiple images having the same requirements.
+
+### Artifact Discovery
+
+If a registry supports referrers API, it should be used for the artifact discovery.
+
+Otherwise, the fallback method on the client said shall be implemented.
+The client:
+
+1. Fetches the image index.
+1. Iterates over descriptors to check if it's compatibility artifact type.
+1. Validates the subject descriptor to check the correct image reference.
```

Comment:
in image index, it includes a manifest list(a list of decriptors), each item will point to image manifest, the subject is usually created in the image manifest. Could the subject be embeded in the item of manifest list, and I am not sure which tool supports this kind of usage

Reply:
You're completely right. Let me update this.